core/src/main/java/com/yahoo/oak/FirstLevelHashArray.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2020, Verizon Media.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.yahoo.oak;
+
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReferenceArray;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+// the array of pointers to HashChunks according to keyHash most significant bits
+class FirstLevelHashArray<K, V> {
+
+    private AtomicReferenceArray<HashChunk<K, V>> chunks;
+    private int msbForFirstLevelHash;
+    private UnionCodec hashIndexCodec;
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
+
+
+    FirstLevelHashArray(int msbForFirstLevelHash, AtomicInteger externalSize, MemoryManager vMM,
+        MemoryManager kMM, OakComparator<K> comparator, OakSerializer<K> keySerializer,
+        OakSerializer<V> valueSerializer, boolean resize) {
```

Comment:
Can you add a comment describing the constructor fields?
Specifically, `msbForFirstLevelHash` and `resize`.

Reply:
fixed, resize parameter removed